FWCore/Framework/interface/EventSetupRecordImpl.h

```diff
@@ -166,23 +163,9 @@ namespace edm {
                              ComponentDescription const*& oDesc,
                              std::shared_ptr<ESHandleExceptionFactory>& whyFailedFactory) const {
         DataKey const* dataKey = nullptr;
-        if (iResolverIndex.value() == std::numeric_limits<int>::max()) {
-          whyFailedFactory = makeESHandleExceptionFactory([this] {
-            NoProductResolverException<DataT> ex(this->key(), {});
-            return std::make_exception_ptr(ex);
-          });
-          iData = nullptr;
-          return;
-        }
```

Comment:
Is this check (and the one below) now effectively done elsewhere?

Reply:
Both of these checks for error conditions duplicate error checks made elsewhere. It is impossible for these error checks to ever fail. This is true both before and after this PR.

We already check here:

https://cmssdt.cern.ch/lxr/source/FWCore/Framework/interface/EventSetupRecord.h#0166

that iResolverIndex does not have an invalid value and that other error check is always run first.

We already check here that pValue is not null:

https://cmssdt.cern.ch/lxr/source/FWCore/Framework/src/ESProductResolver.cc#0082

And that other error check is made first so the one I deleted couldn't ever fail.

It seemed better to me to get rid of that duplication, although I suppose one might argue it's good to check inside that function because it would cause problems if something outside the function changed and either of those conditions started failing... I can put them back in if you would prefer that, although I would have to think a bit about should be in the exception message in those cases.